src/mlpack/core/cv/metrics/r2_score_impl.hpp

```diff
@@ -42,11 +43,21 @@ double R2Score::Evaluate(MLAlgorithm& model,
   // Calculate the denominator i.e.total sum of squares.
   double totalSumSquared = arma::accu(arma::square(responses - meanResponses));
 
-  // Handling undefined R2 Score when both denominator and numerator is 0.0.
-  if (residualSumSquared == 0.0)
-    return totalSumSquared ? 1.0 : DBL_MIN;
-
-  return 1 - residualSumSquared / totalSumSquared;
+  if (AdjustedR2)
+  {
+    // Handling undefined R2 Score when both denominator and numerator is 0.0.
+    if (residualSumSquared == 0.0)
```

Comment:
Should this be `residualSumSquared == 0.0 || totalSumSquared == 0.0`? Because if `totalSumSquared` is 0 the output is undefined.

Reply:
Dear @zoq Thanks for pointing it out. First of all I think, I have placed the `// Handling undefined R2 Score...` part inside the `if (AdjustedR2)` by mistake. It should be above and outside it.

Second, since I do not know what `DBL_MIN` means, I really don't know if `residualSumSquared == 0.0 || totalSumSquared == 0.0` should be put in the if condition. If you could tell me the meaning of `DBL_MIN`, I might be able to help.

Reply:
`DBL_MIN` is the smallest positive normal double, it's an alias for `std::numeric_limits<double>::min` - https://en.cppreference.com/w/cpp/types/numeric_limits/min, hope that helps.

Reply:
Okay, so `AdjustedR2` or `R2` becomes undefined only if `totalSumSquared` is equal to zero. If `residualSumSquared` equals to zero then they are equal to 1. Hence what I would suggest is

if (totalSumSquared = 0)
  return DBL_MIN;
else if (residualSumSquared = 0)
  return 1;

I am checking `totalSumSquared` first just in case if `totalSumSquared` and `residualSumSquared` are both equal to zero, then, the answer should be still undefined. 

I hope it makes sense.

Reply:
It looks like this is correctly handled on lines 46 and 47 now. :+1: